stripecardscan/src/main/java/com/stripe/android/stripecardscan/scanui/SimpleScanStateful.kt

```diff
@@ -0,0 +1,41 @@
+package com.stripe.android.stripecardscan.scanui
+
+import android.util.Log
+import com.stripe.android.camera.framework.AnalyzerLoopErrorListener
+import com.stripe.android.stripecardscan.framework.Config
+
+abstract class ScanState(val isFinal: Boolean)
+
+internal interface SimpleScanStateful<State : ScanState> {
+
+    var scanStatePrevious: ScanState?
+    var scanState: ScanState
+    val scanErrorListener: ScanErrorListener
+
+    fun ensureValidParams(): Boolean
```

Comment:
it looks like this isn't actually referenced anywhere else in this interface, so it doesn't make sense to have it here. Let's either:

a. pull this out into the child classes only
b. give a default implementation in this class that the child classes can override with more checks if needed (e.g. CardImageVerificationActivity)